engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java

```diff
@@ -2178,6 +2178,15 @@ public void execute(String[] args, ConsoleReader reader) throws Exception
 
   }
 
+  private ApplicationReport findApplicationReportFromAppNameOrId(String appNameOrId)
+  {
+    ApplicationReport app = getApplication(appNameOrId);
+    if (app == null) {
+      app = getApplicationByName(appNameOrId);
+    }
+    return app;
```

Comment:
Making `findApplicationReportFromAppNameOrId()` throw a CLIException could 
- on one hand clean up the code so we don't need to rewrite it every time
- on the other hand it would make it inconsistent with getApplication and getApplicationByName, where no exception is thrown. 

Also if we leave the exception out of the method, we would still have the possibility of implementing the logic for "fail all vs. fail only for invalid appName" in the respective commands, which we do not have if we move it into the method